### ISSUE TITLE
Metadata dropout substitutes identity matrices instead of dropping fields

### ISSUE THREAD

**Issue by engichang1467** [engichang1467/Open-Rig3R#39]
## Problem

The paper drops metadata fields so the model learns to cope with their absence. The code
substitutes an identity matrix instead, which teaches the model a specific wrong rig.

§3.4, Embedding Dropout:

> We randomly **drop** each metadata field (camera ID, timestamp, rig pose) with 50%
> probability during training. This structured masking teaches the model to leverage
> metadata when available, and to infer missing context from image content and cross-view
> relationships when it is not.

`datasets/co3d.py:126-139`:

```python
def _maybe_drop_metadata(self, metadata):
    if self.metadata_dropout > 0 and metadata:
        for key in metadata.keys():
            if random.random() < self.metadata_dropout:
                # metadata[key] = None
                if key == 'cam2rig':
                    metadata[key] = torch.eye(4).unsqueeze(0).repeat(self.n_frames, 1, 1)
```

An identity `cam2rig` is not "no information" — it is a valid rig in which every camera
sits at the rig origin with identity rotation. Half of training therefore asserts a false
rig configuration rather than an absent one. The commented-out `metadata[key] = None` is
the behaviour the paper describes.

Note the `None` path is already supported downstream:
`RigAwareTransformerDecoder._collect_metadata_tensors` skips `None` values, and
`_prepare_metadata_tokens` falls back to `learned_meta` when nothing survives.

## Second, unrelated dropout

`models/decoder_transformer.py:86` also applies `nn.Dropout(metadata_dropout)` to the
metadata tokens. That is elementwise feature dropout at p=0.5, a different mechanism from
per-field masking, applied on top. Two dropouts, neither of which is the paper's.

## Third: Waymo has no dropout at all

`datasets/waymo.py:147` returns `{"cam2rig": cam2rig}` unconditionally — `WaymoDataset`
takes no `metadata_dropout` argument, so the primary training dataset never masks
metadata. `scripts/train.py` reads `metadata_dropout` from config but only CO3D uses it.

## Suggested fix

1. Restore the `None` path in `_maybe_drop_metadata` (uncomment, delete the identity
   substitution).
2. Move field dropout into a shared place so Waymo gets it too — both datasets return the
   same `metadata` dict shape, so one helper covers both.
3. Decide what `nn.Dropout(metadata_dropout)` in the decoder is for. If it is a
   duplicate of field dropout, remove it; if it is meant as regularization, decouple its
   rate from the config's `metadata_dropout`.
